### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Markolie**
Fixes https://github.com/ParadiseSS13/Paradise/issues/5782. The button was checking for power, but the holodeck area is never powered.
Fixes https://github.com/ParadiseSS13/Paradise/issues/5785.
Fixes https://github.com/ParadiseSS13/Paradise/issues/3272. Matter eater now checks if a mob is spawnable by tears/gold cores: this prevents matter eater from eating special/boss mobs (especially if Lavaland is ever implemented). Not an ideal solution, but another blacklist would be terrible, and matter eaters being able to eat hostile mobs needs to be looked at anyway.
Fixes https://github.com/ParadiseSS13/Paradise/issues/2668.
Fixes https://github.com/ParadiseSS13/Paradise/issues/5796 (fixed in an earlier patch, including it here to close it). 

Also fixes the checks when ejecting sleepers/cryos so restrained people can't eject them.

🆑 Markolie
bugfix: Fixes the Thunderdome holodeck start buttons not working properly.
bugfix: Fixes ghosts being able to set hydroponic tray lights.
bugfix: Fixes matter eaters being able to eat special hostile mobs such as holoparasites.
bugfix: Fixes programmable unloaders not deconstructing properly.
/🆑

### CODE REVIEW

File: code/game/machinery/Sleeper.dm [hunk 1]
```diff
@@ -454,8 +454,10 @@
 	set name = "Eject Sleeper"
 	set category = "Object"
 	set src in oview(1)
-	if(usr.stat != 0)
+	
+	if(usr.restrained() || usr.stat || usr.weakened || usr.stunned || usr.paralysis || usr.resting) //are you cuffed, dying, lying, stunned or other
```
Comment (Fox-McCloud) [inline code/game/machinery/Sleeper.dm]: wouldn't `if(usr.incapacitated())` be best to check for?

Comment (Markolie) [inline code/game/machinery/Sleeper.dm]: Ah, good catch. I didn't know that function was available. I'll add it now.

**Comment by Fox-McCloud** [comment]
Oh thank goodness, Github made tweaks to their line comment system.

**Comment by Markolie** [comment]
This is ready to go @Fox-McCloud.

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 2]
```diff
@@ -488,7 +488,7 @@
 	return 1
 
 /obj/item/mecha_parts/mecha_equipment/medical/syringe_gun/proc/add_known_reagent(r_id,r_name)
-	if(!(r_id in known_reagents))
+	if(!(r_id in known_reagents) && (r_id in standard_medicines))
```
Comment (TheDZD) [inline code/game/mecha/equipment/tools/medical_tools.dm]: This is probably a bit more of a balance change, might be preferable to just block the chems in [this list](https://github.com/ParadiseSS13/Paradise/blob/master/code/_globalvars/lists/reagents.dm#L50) instead.

Comment (Markolie) [inline code/game/mecha/equipment/tools/medical_tools.dm]: I intended to, but unfortunately enough not all admin chemicals (such as xenomicrobes) are actually in that list.

Comment (Fox-McCloud) [inline code/game/mecha/equipment/tools/medical_tools.dm]: I'd just blacklist those chems for now--we can adjust the list later; it's still a vast improvement over the current "anything goes" list.

**Comment by fludd12** [comment]
I disagree a little with the odysseys change. At the very least, could you have its emag_act remove the chemical loading restriction?

**Comment by Fox-McCloud** [comment]
Merge conflicts--also, again, if you just change the list, then this can be merged.

File: code/game/machinery/cryo.dm [hunk 3]
```diff
@@ -400,6 +401,16 @@
 		var/combined_energy = T20C*current_heat_capacity + air_heat_capacity*air_contents.temperature
 		air_contents.temperature = combined_energy/combined_heat_capacity
 
+/obj/machinery/atmospherics/unary/cryo_cell/proc/expel_gas()
```
Comment (Fox-McCloud) [inline code/game/machinery/cryo.dm]: errrr, this was removed in another PR?

**Comment by Fox-McCloud** [comment]
I went ahead and handled the reagent blacklist in my huge reagent PR; it's more OOP--it's just a `can_synth` var (TG's thing)---it handles all the concerns you had---so just go ahead and remove the mech syringe gun changes and I'll merge this, post-haste.

**Comment by Markolie** [comment]
All done!

**Pull request merged**